### ISSUE TITLE
Creating a diagram 

### ISSUE THREAD

**Issue by bnmounir** [open-learning-exchange/open-learning-exchange.github.io#2186]
### Problem (more of a suggestion)

If you think it's a good idea I would love to hear from you, what would be the most important commands that should be listed here and I will work on it. (also if you have a better idea about how it should be designed, for better understanding)

### Screenshots

<img width="780" alt="diag" src="https://user-images.githubusercontent.com/40181209/53119780-3bcb9700-351e-11e9-9012-f1680c58252f.png">

### Proposed solution

adding a diagram in GitHub steps listing the commands needed for most ops, to make it easy for junior devs to navigate the project.

**Comment by singharpita** [comment]
@bnmounir It will be good to have git workflow we follow instead of command flow.

**Comment by lmmrssa** [comment]
@bnmounir Yes as @singharpita said you. It will be nice to have block diagram of workflow. We want people to learn commands and also workflow. If we put commands on image they will just copy paste it.

**Comment by bnmounir** [comment]
hi @lmmrssa @singharpita, I see your point, I was just trying to come up with a picture showing the data flow, especially people without a lot of experience such as myself, I drew a little diagram to clear my confusion (what's on the computer VS what's on Github, also how to sync everything to avoid conflicts) and I thought it might be helpful for others.

**Comment by lmmrssa** [comment]
Sure @bnmounir that's why we wanted you to create block diagram of gitflow. Command is something they have to learn and understand not just copy paste. So flow diagram is something you can come up with

**Comment by bnmounir** [comment]
hey @lmmrssa what do you think of this one, I think it draws a mental image of what's happening. 

<img width="684" alt="diag2" src="https://user-images.githubusercontent.com/40181209/53355870-0ba63e80-38f8-11e9-8ebb-ff997b86ae5e.png">

**Comment by lmmrssa** [comment]
@bnmounir  I don't get the concept of State1 and State2.
Because you are making fork from State1 creating branch then sending that request as State2, which looks like you wanted to say State is two versions of code. But later you are fetching from state2.
Also pull request is from work branch to upstream, while looking at image it looks like it goes to master branch and then from there go to upstream.

If you are familiar with flowchart concept then making one that shows all steps will be nice.

**Comment by bnmounir** [comment]
@lmmrssa I hear you, it would be really helpful if you could please list a few words you would like to see in this workflow, and I'll work on a sketch.

**Comment by lmmrssa** [comment]
@bnmounir words will be something similar. But just the way of presentation needs to be changed.
Such as
* Fork repository
* Create new branch for issue from master branch
* Commit changes
* Create pull request
* Sync fork repository
But all these needs to be shown as flowchart and there could also include other stuff not mentioned here.

**Comment by zelmi** [comment]
I think the idea of a diagram is great! I would suggest making the theme color the blue that is already  used in the tutorial notes. That way the page will look more seamless. For example:

<img width="844" alt="Screen Shot 2019-03-25 at 2 57 46 PM" src="https://user-images.githubusercontent.com/29234807/54946493-8de14d00-4f0e-11e9-8a33-a3fe01b16dfd.png">

This might help: 
<img width="193" alt="Screen Shot 2019-03-25 at 3 03 58 PM" src="https://user-images.githubusercontent.com/29234807/54946786-3e4f5100-4f0f-11e9-9338-c22840ceeb8d.png">

**Comment by SNutakki** [comment]
I think this might still be a good idea if someone wants to work on it. It depends on what exactly we're including in this diagram though, as there are some diagrams and explanations in the tutorial already.

**Comment by hiroTochigi** [comment]
How about these drawings?
Do they help?

1. Fork and rename
![git1](https://user-images.githubusercontent.com/44413241/59558783-c77e9d00-8fbf-11e9-9d44-9687de307ba1.jpg)




2. Create a branch
![git2](https://user-images.githubusercontent.com/44413241/59558784-cc435100-8fbf-11e9-8aec-3f94c662159f.jpg)




3. Change content and Pull Request
![git3](https://user-images.githubusercontent.com/44413241/59558788-eb41e300-8fbf-11e9-9180-b09c4e3c2b0f.jpg)


4. Pull request is approved and your branch is merged 
![git4](https://user-images.githubusercontent.com/44413241/59558792-09a7de80-8fc0-11e9-8635-8ca99d849d9a.jpg)

**Comment by pattanawadee88** [comment]
I like the idea! Diagrams are much easier to understand than words.
